src/oracle/connection/oracle_value.rs

```diff
@@ -75,15 +75,15 @@ impl<'a> OracleValue<'a> {
             Raw {
                 tpe: oracle::sql_type::OracleType::Number(prec, 0),
                 ..
-            } if prec == 5 => OciDataType::SmallInt,
+            } if prec <= 5 => OciDataType::SmallInt,
             Raw {
                 tpe: oracle::sql_type::OracleType::Number(prec, 0),
                 ..
-            } if prec == 10 => OciDataType::Integer,
+            } if prec <= 10 => OciDataType::Integer,
             Raw {
                 tpe: oracle::sql_type::OracleType::Number(prec, 0),
                 ..
-            } if prec == 19 => OciDataType::BigInt,
+            } if prec <= 19 => OciDataType::BigInt,
```

Comment:
Could we map anything with `scale == 0` that wasn't matched before to `BigInt`? I know the number might then not fit into an `i64`, but it would make it way more usable for external tables, since `number(38, 0)` or `number(*, 0)` are commonly used for integer / id columns.